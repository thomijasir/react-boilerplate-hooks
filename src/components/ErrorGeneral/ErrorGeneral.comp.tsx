import React, { FC } from 'react';
import ERROR_DEFAULT from '../../assets/images/general_error_default.png';
import packageInfo from '../../../package.json';
import './ErrorGeneral.css';

export interface IErrorGeneralProps {
  title?: string;
  message?: string;
  image?: string;
  isError: boolean;
}

export const ErrorGeneralDefaultProps: IErrorGeneralProps = {
  title: 'Something Wrong!',
  message: 'Opps, failure open apps or your session has expired.',
  image: ERROR_DEFAULT,
  isError: false,
};

export const ErrorGeneralNamespace: string = 'ErrorGeneral';

const ErrorGeneral: FC<IErrorGeneralProps> = ({
  title,
  message,
  image,
  isError,
}) => {
  return (
    <div
      className={`error-general transition duration-150 ease-in-out ${
        isError ? 'on' : ''
      }`}
    >
      <div className="error-container">
        <img src={image} alt="errorPage" />
        <h2 className="error-title">{title}</h2>
        <p className="error-desc">{message}</p>
      </div>
      <div className="error-footer">
        <p className="text-sm">v{packageInfo.version}</p>
      </div>
    </div>
  );
};

ErrorGeneral.displayName = ErrorGeneralNamespace;
ErrorGeneral.defaultProps = ErrorGeneralDefaultProps;

export default React.memo(ErrorGeneral);
