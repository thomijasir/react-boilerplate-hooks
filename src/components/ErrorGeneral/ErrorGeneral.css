.error-general {
  position: fixed;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  text-align: center;
  opacity: 0;
  @apply bg-white;
}

.error-general .error-container {
  width: 100%;
  text-align: center;
  max-width: 20rem;
  margin: auto;
}

.error-general .error-container img {
  width: 10rem;
  height: 10rem;
  margin: auto
}

.error-general .error-container .error-title {
  @apply text-lg font-bold mt-4;
}

.error-general .error-container .error-desc {
  @apply text-base;
}

.error-general.on {
  opacity: 1;
  z-index: 9998;
}

.error-general .error-footer {
  width: 100%;
  position: absolute;
  bottom: 0.08rem;
  @apply text-gray-400 p-4;
}

.error-general .error-action {
  border-top: .0625rem solid #000;
  padding: 0.08rem 0.16rem 0 0.16rem;
  box-sizing: border-box;
}