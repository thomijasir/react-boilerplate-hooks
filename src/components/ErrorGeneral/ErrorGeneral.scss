@import '../../assets/style/colors';
@import '../../assets//style/var';
.error-general {
  position: fixed;
  z-index: -1;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: $white;
  display: flex;
  align-items: center;
  text-align: center;
  opacity: 0;
  .error-container {
    margin: auto;
    max-width: 320px;
    padding: 0 16px;
    img {
      width: 160px;
      height: 160px;
    }
    .error-title {
      font-weight: $font-w-semi-bold;
      font-family: $font-second;
      font-size: 18px;
    }
    .error-desc {
      font-size: 16px;
      line-height: 1.5;
      color: $soft-grey;
    }
  }
  &.on {
    opacity: 1;
    z-index: 9998;
  }
  .error-footer {
    width: 100%;
    position: absolute;
    bottom: 0.08rem;
  }
  .error-action {
    border-top: 1px solid $very-light-grey;
    padding: 0.08rem 0.16rem 0 0.16rem;
    box-sizing: border-box;
  }
  .version-info {
    font-size: 0.08rem;
    opacity: 0.5;
    margin-top: 0.02rem;
    margin-bottom: 0.1rem;
  }
}
