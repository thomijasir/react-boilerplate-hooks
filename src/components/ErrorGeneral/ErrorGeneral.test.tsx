import React from 'react';
import render from 'react-test-renderer';
import ErrorGeneral from './ErrorGeneral.comp';

afterEach(() => {
  jest.clearAllMocks();
});

describe('Error General Component Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const compProps = {
        title: 'Hello World',
        message: 'Sample Word',
        isError: false,
      };
      const wrapper = render.create(<ErrorGeneral {...compProps} />);
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
