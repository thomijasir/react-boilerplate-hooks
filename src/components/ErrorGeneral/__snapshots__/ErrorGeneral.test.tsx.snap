// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error General Component Snap Test render() renders the component 1`] = `
<div
  className="error-general transition duration-150 ease-in-out "
>
  <div
    className="error-container"
  >
    <img
      alt="errorPage"
      src="general_error_default.png"
    />
    <h2
      className="error-title"
    >
      Hello World
    </h2>
    <p
      className="error-desc"
    >
      Sample Word
    </p>
  </div>
  <div
    className="error-footer"
  >
    <p
      className="text-sm"
    >
      v
      0.0.1
    </p>
  </div>
</div>
`;
